numpy/core/fromnumeric.py

```diff
@@ -837,12 +837,12 @@ def argpartition(a, kth, axis=-1, kind='introselect', order=None):
     return _wrapfunc(a, 'argpartition', kth, axis=axis, kind=kind, order=order)
 
 
-def _sort_dispatcher(a, axis=None, kind=None, order=None):
+def _sort_dispatcher(a, axis=None, kind=None, order=None, by=None):
     return (a,)
 
 
 @array_function_dispatch(_sort_dispatcher)
-def sort(a, axis=-1, kind=None, order=None):
+def sort(a, axis=-1, kind=None, order=None, by=None):
```

Comment:
If we're following https://github.com/numpy/numpy/pull/16700/files#r460500656 we should also

```suggestion
def sort(a, axis=-1, kind=None, order=None, *, by=None):
```